src/main/java/sample/dao/ProductDAO.java

```diff
@@ -5,11 +5,12 @@
 import java.util.List;
 
 public interface ProductDAO {
-    List<Product> get();
+    List<Product> getAllProducts();
 
-    void add();
+    void add(Product product);
 
+    //delete by ID or by Name ???
```

Comment:
In your case you can delete by id. 
Because you are using the entire Product object.